### PR TITLE
feat: provide next affiliation ID and expert panel ID

### LINKED ISSUE #133: Add "Next Affiliation ID" functionality
Populate the next affiliation ID based on what is in the DB for the user.
Populate Expert Panel ID based on Type selected and Affiliation ID.

This is make it easier for user to add affiliations and not have to already know what the next ID number should be.

### PR DESCRIPTION
**Pull request by sanchegm**
Description
JS:
- Add custom JS functionality to hide Affiliation ID and Expert Panel ID input fields from user when creating a new affiliation and show these values when editing an affiliation. Based on documentation: https://docs.djangoproject.com/en/5.1/ref/contrib/admin/javascript/
- Add custom JS functionality to hide CDWG field when Somatic Cancer-VCEP is selected at SC-VCEP type will always have Somatic Cancer CDWG.

Next Affiliation functionality:
- Add functionality to calculate next available Affiliation ID if New based on last largest Affil ID in DB. 
- Add functionality to calculate EP ID based on Type and Affiliation ID.

Misc:
- Add thread safety and atomic database interactions based on documentation: https://docs.djangoproject.com/en/5.1/topics/db/transactions/#module-django.db.transaction
- Update clean method in admin.py to validate user input information.
- Update tests in test.py to test for updated validation errors.

Issue Ticket Number and Link 
Closes #133 

Checklist
 Remove any options that are not applicable
 [X ] I have reviewed the [how-to guide](https://github.com/ClinGen/stanford-affils/pull/how-to.md) and the [contributing file](https://github.com/ClinGen/stanford-affils/CONTRIBUTING.md).
 [X] I have run required [code checks](https://github.com/ClinGen/stanford-affils/pull/how-to.md#run-code-checks) and resolved any blockers.
 [X] I have created the necessary [tests](https://github.com/ClinGen/stanford-affils/src/app_test.py) to show my changes are effective and work as intended.
 [X] I have thoroughly commented my code, especially in more complex areas.
 [X] I have updated any necessary documentation.

Screenshots / Recordings
Screenshot 2024-09-24 at 1 43 21 PM

### CODE REVIEW

**Comment by sanchegm** [comment]
Refactor based on conversations from Matt Wright in slack about SC-VCEPs.

**Review by liammulh — COMMENTED** [review]
I think this looks good, @sanchegm. I read through the Django docs on [database transactions](https://docs.djangoproject.com/en/5.1/topics/db/transactions/#module-django.db.transaction), and I'm wondering what you think about setting `ATOMIC_REQUESTS` to `True`. Relevant docs [here](https://docs.djangoproject.com/en/5.1/topics/db/transactions/#tying-transactions-to-http-requests).

File: src/affiliations/admin.py [hunk 1]
```diff
@@ -200,6 +208,11 @@ class AffiliationsAdmin(ModelAdmin):
 
     form = AffiliationForm
 
+    # class Media:
```
[the remark this answers is not in the document]
Comment (sanchegm) [reply src/affiliations/admin.py:211]: That is a mistake, thank you for catching! Fixed.

File: src/main/settings.py [hunk 2]
```diff
@@ -38,6 +38,7 @@
 
 CSRF_TRUSTED_ORIGINS = ["https://affils.clinicalgenome.org"]
 
+ATOMIC_REQUESTS = True
```
Comment (liammulh) [inline src/main/settings.py:41]: I think this should go in the `DATABASES['default']` block.

Comment (sanchegm) [reply src/main/settings.py:41]: You are correct! I read the documentation incorrectly. Fixed!

**Review by liammulh — APPROVED** [review]

**Pull request merged**